pyccel/ast/builtin_objects/list_functions.py

```diff
@@ -0,0 +1,39 @@
+# coding: utf-8
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+The List container has a number of built-in methods that are 
+always available.
+
+In this module we implement List methods.
+"""
+
+from pyccel.ast.internals import PyccelInternalFunction
+from pyccel.ast.datatypes import NativeHomogeneousList
+
+class ListPop(PyccelInternalFunction) :
+    """
+    Represents a call to the .pop() method.
+    
+    Represents a call to the .pop() method wich 
+    removes the item at the specified index. 
+    The method also returns the removed item.
+    
+    Parameters
+    ----------
+    *args : iterable
+        The arguments passed to the function call.
+    """
+    __slots__ = ('_dtype',)
+    _precision = -1
+    _rank  = 0
+    _shape = None
+    _order = None
+    _class_type = NativeHomogeneousList()
+    name = 'pop'
+    def __init__(self, *args):
+        super().__init__(*args)
+        self._dtype = args[0].dtype
```

Comment:
As described in the second paragraph [here](https://github.com/pyccel/pyccel/blob/devel/developer_docs/ast_nodes.md#pyccel-internal-function), objects inheriting from PyccelInternalFunction must provide a constructor whose positional and keyword arguments match the names of the positional and keyword arguments of the Python function being described.

According to [Python's docs](https://docs.python.org/3/tutorial/datastructures.html#more-on-lists) list.append only accepts 1 argument whose name may be `i`. Your function takes any number of arguments. What are these additional arguments supposed to represent? Further the argument to `pop` is optional however this case does not seem to be handled by your implementation. This isn't a problem in the Python implementation but it will be much simpler for the C/Fortran implementations if it is already defined here.